web/TrackPlayer/PlaylistPlayer.ts

```diff
@@ -20,28 +20,30 @@ export class PlaylistPlayer extends Player {
   }
 
   protected async onTrackEnded() {
-    switch (this.repeatMode) {
-      case RepeatMode.Track:
-        if (this.currentIndex !== undefined) {
-          await this.goToIndex(this.currentIndex);
-        }
-        break;
-      case RepeatMode.Playlist:
-        if (this.currentIndex === this.playlist.length - 1) {
-          await this.goToIndex(0);
-        }
-        break;
-      default:
-        try {
-          await this.skipToNext();
-        } catch (err) {
-          if ((err as Error).message !== 'playlist_exhausted') {
-            throw err;
+    try {
```

Comment:
I see that this try/catch is to catch the `playlist_exhaused` error since we're calling another `skipToNext`. However, I'm not convinced that, in `RepeatMode.Playlist`, we should be catching that error. If the whole playlist should be repeated, I can't envision a scenario where `skipToNext` should properly throw that error, and if it does, it seems something strange must've happened (unless I'm missing a use-case as I'm thinking through it).

Is there something I'm not seeing there?

Reply:
Ow yeah, I think you are right.

Reply:
Beside that however, if the playlist repeats, should this.onPlaylistEnded(); be called?

Reply:
And what happends if the playlist is emptied?

Reply:
Can you look at how Swift and KotlinAudio handle that? I don't think `onPlaylistEnded` should be called, but we'll want feature parity with those.

Reply:
I don't think it does. I just committed the smaller try-catch!